### DISCUSSION TITLE
Is there a way to "refresh" a `PerspectiveWidget` after it's underlying `Table` has been updated?

### DISCUSSION THREAD

**Discussion by dhirschfeld** [perspective-dev/perspective#1490] [Q&A] [answered]
In JupyterLab when I call `psp.Table.replace(new_data: pd.DataFrame)` to update a table with new data, the  `PerspectiveWidget` referencing the table isn't updated to reflect the changes until I call `display`. If I call `display` on the widget again it resets the state of the sidebar which I'd like to avoid.

Is the some way to get the `PerspectiveWidget` to reflect, in the UI, changes to it's underlying `Table` (short of calling `display` again)?

Is it a bug that the toggle state of the sidebar is reset when `display` is called?

**Comment by sc1f** [comment, marked as answer]
Currently replace and clear don’t fully work in the PerspectiveWidget (which uses client/server mode). You can pass server=True to the widget constructor and it should work.

On Jul 26, 2021, at 11:02 AM, Dave Hirschfeld ***@***.***> wrote:

﻿

In JupyterLab when I call psp.Table.replace(new_data: pd.DataFrame) to update a table with new data, the PerspectiveWidget referencing the table isn't updated to reflect the changes until I call display. If I call display on the widget again it resets the state of the sidebar which I'd like to avoid.

Is the some way to get the PerspectiveWidget to reflect, in the UI, changes to it's underlying Table (short of calling display again)?

Is it a bug that the toggle state of the sidebar is reset when display is called?

**Comment by dhirschfeld** [comment]
Works like charm - thanks @sc1f!
